nixos/modules/services/web-apps/stirling-pdf.nix

```diff
@@ -0,0 +1,140 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    getExe
+    literalExpression
+    mkEnableOption
+    mkIf
+    mkMerge
+    mkOption
+    mkPackageOption
+    optional
+    ;
+
+  inherit (lib.types)
+    attrsOf
+    nullOr
+    path
+    port
+    str
+    submodule
+    ;
+
+  cfg = config.services.stirling-pdf;
+in
+
+{
+  meta.maintainers = with lib.maintainers; [ thubrecht ];
+
+  options.services.stirling-pdf = {
+    enable = mkEnableOption "Stirling-PDF, a web application for manipulating PDF files";
+
+    package = mkPackageOption pkgs "stirling-pdf" { };
+
+    domain = mkOption {
+      type = str;
+      description = ''
+        The domain for the application.
+      '';
+    };
+
+    port = mkOption {
+      type = port;
+      default = 8080;
+      description = ''
+        Port that Stirling-PDF will listen on.
+      '';
+    };
+
+    environment = mkOption {
+      type = attrsOf str;
+      defaultText = ''
+        {
+          SERVER_PORT = cfg.port;
+        }
+      '';
+      description = ''
+        Settings for the stirling-pdf app.
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = nullOr path;
+      default = null;
+      description = ''
+        File containing additional environment variables to pass to Stirling PDF.
+      '';
+    };
+
+    nginx = mkOption {
+      type = nullOr (submodule (import ../web-servers/nginx/vhost-options.nix { inherit config lib; }));
+      default = null;
+      description = ''
+        With this option, you can customize the nginx virtualHost settings.
+      '';
+      example = literalExpression ''
+        {
+          # To enable encryption and let Let's Encrypt take care of certificate
+          forceSSL = true;
+          enableACME = true;
+        }
+      '';
```

Comment:
Given that this is a very simple service to reverse proxy (i.e: it's just a port), I don't think we need an explicit option in the module for it.

Reply:
Well, it cost nothing to have it and doesn't force you to use it (?).

Reply:
(except maintenance for the module maintainer, but well...)

Reply:
Well, better safe than sorry